src/player.cpp

```diff
@@ -12228,26 +12228,36 @@ int player::encumb( body_part bp ) const
 int player::encumb(body_part bp, double &layers, int &armorenc) const
 {
     int ret = 0;
-    int layer[MAX_CLOTHING_LAYER] = { };
+    // First is the total of encumbrance on the given layer,
+    // and second is the highest encumbrance of any one item on the layer.
+    std::array<std::pair<int, int>, MAX_CLOTHING_LAYER / 10> layer = { };
 
     for( auto& w : worn ) {
         if( !w.covers(bp) ) {
             continue;
         }
+        std::pair<int, int> &this_layer = layer[w.get_layer()];
```

Comment:
changing this to `std::pair<int, int> &this_layer = layer[w.get_layer() / 10];` resolves the out of bounds error.